### PR TITLE
React: Fix issue with react 18 implementation

### PR DESCRIPTION
**Pull request by tmeasday**
Follow up on https://github.com/storybookjs/storybook/pull/19118

Issue: When implementing a rendering story that checks a new button isn't created when setting an arg, I realised that I had actually broken things in https://github.com/storybookjs/storybook/pull/18737.

Now we have this story, we shouldn't be able to break things in this way again 🤞 

## What I did

Changed the react callback behaviour

## How to test

1. Run rendering story
2. Also check that the `useEffect` helper still works (there's a `hooks.stories.ts` story for this)
3. The docs E2E should pick up if the callback isn't getting called properly.

### CODE REVIEW

File: code/renderers/react/src/render.tsx [hunk 1]
```diff
@@ -160,6 +156,7 @@ export async function renderToDOM(
   // This could leads to issues like below:
   // https://github.com/storybookjs/react-storybook/issues/81
   // (This is not the case when we change args or globals to the story however)
+  console.log({ forceRemount });
```
Comment (shilman) [inline code/renderers/react/src/render.tsx:159]:
```suggestion
```

**Review by shilman — APPROVED** [review]
Love it. Tests FTW!!!

**Comment by shilman** [comment]
@tmeasday what's going on with CI here?

**Comment by tmeasday** [comment]
Oh, it needs to be updated with this commit: https://github.com/storybookjs/storybook/pull/19105/commits/1cb2f9dfd89bf9b3d1e6a3c6ea128fc9a6cc23ff which is on the parent branches but hasn't made it in here yet.

**Comment by IanVS** [comment]
Since https://github.com/storybookjs/storybook/pull/18737 was just released in 6.5.11, should this fix be patched back to 6.5 as well?

**Comment by tmeasday** [comment]
Yes!

**Comment by jmaahir** [comment]
Very good job 👍

**Pull request merged**